Ryujinx.HLE/HOS/Services/Nv/NvDrvServices/NvHostChannel/NvHostChannelDeviceFile.cs

```diff
@@ -0,0 +1,352 @@
+                        break;
+                }
+            }
+            else if (command.Type == NvIoctl.NvHostMagic)
+            {
+                switch (command.Number)
+                {
+                    case 0x01:
+                        result = CallIoctlMethod<int>(SetNvMapFd, arguments);
+                        break;
+                    case 0x03:
+                        result = CallIoctlMethod<uint>(SetTimeout, arguments);
+                        break;
+                    case 0x08:
+                        result = SubmitGpfifo(arguments);
+                        break;
+                    case 0x09:
+                        result = CallIoctlMethod<AllocObjCtxArguments>(AllocObjCtx, arguments);
+                        break;
+                    case 0x0b:
+                        result = CallIoctlMethod<ZcullBindArguments>(ZcullBind, arguments);
+                        break;
+                    case 0x0c:
+                        result = CallIoctlMethod<SetErrorNotifierArguments>(SetErrorNotifier, arguments);
+                        break;
+                    case 0x0d:
+                        result = CallIoctlMethod<NvChannelPriority>(SetPriority, arguments);
+                        break;
+                    case 0x18:
+                        result = CallIoctlMethod<AllocGpfifoExArguments>(AllocGpfifoEx, arguments);
+                        break;
+                    case 0x1a:
+                        result = CallIoctlMethod<AllocGpfifoExArguments>(AllocGpfifoEx2, arguments);
+                        break;
+                    case 0x1d:
+                        result = CallIoctlMethod<uint>(SetTimeslice, arguments);
+                        break;
+                }
+            }
+            else if (command.Type == NvIoctl.NvGpuMagic)
+            {
+                switch (command.Number)
+                {
+                    case 0x14:
+                        result = CallIoctlMethod<ulong>(SetUserData, arguments);
+                        break;
+                }
+            }
+
+            return result;
+        }
+
+        private NvInternalResult Submit(Span<byte> arguments)
+        {
+            int                 headerSize           = Unsafe.SizeOf<SubmitArguments>();
+            SubmitArguments     submitHeader         = MemoryMarshal.Cast<byte, SubmitArguments>(arguments)[0];
+            Span<CommandBuffer> commandBufferEntries = MemoryMarshal.Cast<byte, CommandBuffer>(arguments.Slice(headerSize)).Slice(0, submitHeader.CmdBufsCount);
+            NvGpuVmm            vmm                  = NvHostAsGpuDeviceFile.GetAddressSpaceContext(_owner).Vmm;
+
+            foreach (CommandBuffer commandBufferEntry in commandBufferEntries)
+            {
+                NvMapHandle map = NvMapDeviceFile.GetMapFromHandle(_owner, commandBufferEntry.MemoryId);
+
+                int[] cmdBufData = new int[commandBufferEntry.WordsCount];
```

Comment:
```suggestion
                int[] commandBufferData = new int[commandBufferEntry.WordsCount];
```
Since there is `commandBufferEntry` maybe it's better to use the same for the var name.